singlem/pipe.py

```diff
@@ -537,6 +626,36 @@ def assign_taxonomy_and_process(self, **kwargs):
         #### Process taxonomically assigned reads
         otu_table_object = OtuTable()
         package_to_taxonomy_bihash = {}
+
+
+        # This code snippet counts how many hits each read has.
+        # Can be used to do read_length corrections for multihit nanopore reads.
+        # Other snippets need to be uncommented in the 
+        # UnalignedAlignedNucleotideSequence class in sequence_classes.py
+        
+        # hits_per_true_name = []
+        # read_to_true_name = {}
+        # for readset in extracted_reads:
+        #     for sequence in readset.unknown_sequences:
+        #         true_name = sequence.name.split('~')[0]
+        #         read_to_true_name[sequence.name] = true_name
+        #         hits_per_true_name.append(true_name)
+
+        # from collections import Counter
+        # hits_per_true_name = Counter(hits_per_true_name)
+
+        # hits_per_read = {}
+        # for false_name, true_name in read_to_true_name.items():
+        #     if true_name in hits_per_true_name:
+        #         hits_per_read[false_name] = hits_per_true_name[true_name]
+
+        # for readset in extracted_reads:
+        #     for sequence in readset.unknown_sequences:
+        #         if sequence.name in hits_per_read:
+        #             sequence.num_hits_on_read = hits_per_read[sequence.name]
+        #             # print(sequence.name, sequence.num_hits_on_read)
+        
+
```

Comment:
As above, I'd remove this, but keep handy in your notes or elsewhere if needed in future.